### PR TITLE
fix props validation fails (#3183)

### PR DESCRIPTION
**Pull request by kazupon**

### CODE REVIEW

File: src/compiler/compile-props.js [hunk 1]
```diff
@@ -122,6 +123,9 @@ export function compileProps (el, propOptions, vm) {
     } else if ((value = getAttr(el, attr)) !== null) {
       // has literal binding!
       prop.raw = value
+    } else if (propsData && (value = propsData[attr]) !== null) {
```
Comment (yyx990803) [inline src/compiler/compile-props.js]: `attr` is hyphenated, but the user may use camelCase prop names in `propsData`. We need to check  both `propsData[attr]` and `propsData[name]`.

Comment (iagafonov) [inline src/compiler/compile-props.js]: Maybe user can **only** use camelCase propsData[path]?

Comment (yyx990803) [inline src/compiler/compile-props.js]: True, that makes more sense. then we should just replace `attr` with `name` here.

**Comment by kazupon** [comment]
Thanks review!
I fixed.

**Pull request merged**